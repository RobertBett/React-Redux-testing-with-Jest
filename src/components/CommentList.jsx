import React from 'react';
import { connect } from 'react-redux';


class CommentList extends React.Component{
    renderComments(){
        return this.props.comments.slice(0).reverse().map( comment =>{
            return <li key={comment}>{comment}</li>
        })
    }
    render(){
        return(
            <div>
                <ul>
                    {this.renderComments()}
                </ul>
            </div>
        )
    }
}

const mapStateToProps = state =>({
    comments:state.comments
});

export default connect(mapStateToProps)(CommentList);
