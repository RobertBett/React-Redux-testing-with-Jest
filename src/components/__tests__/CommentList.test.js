import React from 'react'
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import CommentList from '../CommentList';
import Root from '../../Root';



describe('Comment List component', () => {
    let wrapped;

    beforeEach(() =>{
        const initialState ={
            comments: ['comment', 'comment1','blah blah']
        }
        wrapped = mount(
        <Root initialState={initialState}>
            <CommentList/>
        </Root>);
    });
    it('creates one li per Comment',()=>{
        expect(wrapped.find('li').length).toEqual(3);
    })
    it('shows the text ', () =>{
        expect(wrapped.render().text()).toContain('comment')
        expect(wrapped.render().text()).toContain('comment1')
        expect(wrapped.render().text()).toContain('blah blah')
    })

    it('should match the snapshot', () =>{
        expect(toJSON(wrapped)).toMatchSnapshot()
    })
})
