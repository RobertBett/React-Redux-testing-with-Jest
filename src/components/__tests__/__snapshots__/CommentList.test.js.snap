// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment List component should match the snapshot 1`] = `
<Component
  initialState={
    Object {
      "comments": Array [
        "comment",
        "comment1",
        "blah blah",
      ],
    }
  }
>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <Connect(CommentList)>
      <CommentList
        comments={
          Array [
            "comment",
            "comment1",
            "blah blah",
          ]
        }
        dispatch={[Function]}
      >
        <div>
          <ul>
            <li
              key="blah blah"
            >
              blah blah
            </li>
            <li
              key="comment1"
            >
              comment1
            </li>
            <li
              key="comment"
            >
              comment
            </li>
          </ul>
        </div>
      </CommentList>
    </Connect(CommentList)>
  </Provider>
</Component>
`;
